diff --git a/dist/index.d.ts b/dist/index.d.ts
index c46b4ca85a24c07c86015ac14b80a3ee1dc3c346..8550e86dd5e541477378ed0065bbd23885b94fb1 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,5 +1,4 @@
 /// <reference types="node" />
-import Bluebird from 'bluebird';
 import * as events from 'events';
 import webpack from 'webpack';
 type FilePath = string;
@@ -21,7 +20,7 @@ interface FileEvent extends events.EventEmitter {
  * Cypress asks file preprocessor to bundle the given file
  * and return the full path to produced bundle.
  */
-type FilePreprocessor = (file: FileEvent) => Bluebird<FilePath>;
+type FilePreprocessor = (file: FileEvent) => Promise<FilePath>;
 type WebpackPreprocessorFn = (options: PreprocessorOptions) => FilePreprocessor;
 /**
  * Cypress file preprocessor that can bundle specs
diff --git a/dist/lib/utils.d.ts b/dist/lib/utils.d.ts
index f3fcfcbaf51199c9af5ac6db84410427c0aa03d1..1511e71f9de1869cee66deb19636eb75ac37434e 100644
--- a/dist/lib/utils.d.ts
+++ b/dist/lib/utils.d.ts
@@ -1,8 +1,7 @@
-import Bluebird from 'bluebird';
 declare function createDeferred<T>(): {
     resolve: (thenableOrResult?: T | PromiseLike<T> | undefined) => void;
     reject: any;
-    promise: Bluebird<T>;
+    promise: Promise<T>;
 };
 declare const _default: {
     createDeferred: typeof createDeferred;
