diff --git a/dist/cjs/createStartHandler.cjs b/dist/cjs/createStartHandler.cjs
index 4f5611e76e7cad308fb1403aa880a2718294ce7d..53245d60847dd2bcc205060ddec738f8bb60a135 100644
--- a/dist/cjs/createStartHandler.cjs
+++ b/dist/cjs/createStartHandler.cjs
@@ -16,7 +16,7 @@ function createStartHandler({
       const history = reactRouter.createMemoryHistory({
         initialEntries: [href]
       });
-      const router = createRouter();
+      const router = createRouter(request);
       ssrServer.attachRouterServerSsrUtils(router, getRouterManifest == null ? void 0 : getRouterManifest());
       router.update({
         history
diff --git a/dist/esm/createStartHandler.js b/dist/esm/createStartHandler.js
index 006192a4cf55482cb61cf40a726b1076718f2b35..dc49a78783716210c398d3988cecf46bbcc53115 100644
--- a/dist/esm/createStartHandler.js
+++ b/dist/esm/createStartHandler.js
@@ -14,7 +14,7 @@ function createStartHandler({
       const history = createMemoryHistory({
         initialEntries: [href]
       });
-      const router = createRouter();
+      const router = createRouter(request);
       attachRouterServerSsrUtils(router, getRouterManifest == null ? void 0 : getRouterManifest());
       router.update({
         history
